### PR TITLE
Add a quick_look feature of interactive lightcurve and tpf with slider

### PR DESCRIPTION
**Pull request by gully**
Relates to Issue #98  

Work in progress-- still brittle due to assumptions of "typical" TPFs, etc.  many edge/corner cases to be considered.  Stylistic improvements also pending.

### CODE REVIEW

**Comment by gully** [comment]
20180409_lightkurve_quicklook

**Comment by christinahedges** [comment]
@gully ++++++++++++++++++

Such a nice job

File: lightkurve/targetpixelfile.py [hunk 1]
```diff
@@ -608,6 +608,82 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
                                                        alpha=.6))
         return ax
 
+    def quick_look(self, lc=None):
```
Comment (barentsen) [inline lightkurve/targetpixelfile.py]: Let's go with `interact()`?

**Comment by barentsen** [comment]
We probably don't need both the pause and stop button?  In fact we may be able to simplify the interface by having a single "pause/start" button?

File: lightkurve/targetpixelfile.py [hunk 2]
```diff
@@ -608,6 +608,82 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
...
+            from ipywidgets import interact
+            import ipywidgets as widgets
+            from bokeh.io import push_notebook, show, output_notebook
+            from bokeh.plotting import figure
+            from bokeh.models import Span
+            from bokeh.layouts import row
+            output_notebook()
+        except ImportError:
+            raise ImportError('The quicklook tool requires Bokeh and ipywidgets.  See the Installation Guide.')
+            return
```
Comment (mirca) [inline lightkurve/targetpixelfile.py]: I think the ``return`` statement is not needed

File: lightkurve/targetpixelfile.py [hunk 3]
```diff
@@ -608,6 +608,82 @@ def plot(self, ax=None, frame=0, cadenceno=None, bkg=False, aperture_mask=None,
...
+
+        s2 = figure(plot_width=300, plot_height=300, title='Target Pixel File')
+        s2.yaxis.axis_label = 'Pixel Row Number'
+        s2.xaxis.axis_label = 'Pixel Column Number'
+
+        s2_dat = s2.image([self.flux[0,:,:]], x=self.column, y=self.row,
+                          dw=self.shape[2], dh=self.shape[1], dilate=True)#, palette="Spectral11")
+
+
+
```
Comment (mirca) [inline lightkurve/targetpixelfile.py]: I would remove those blank lines

**Comment by gully** [comment]
whoops, my hotfix for issue #109 is failing tests because cadenceno did not previously appear in the synthetic test lightcurves.  other than that, the interactive widget should be near a minimum viable product.  user testing is welcomed.

**Comment by gully** [comment]
Ok, I'm updating this pull request with a few comments from Geert from last Friday.
- [ ] Make the slider bar be the provided "Cadence Number", not "index"

**Comment by gully** [comment]
OK, I've done all of the UI improvements from @barentsen!  I think we should get a little extra user testing and then merge this in to master, yeah?

It's failing one test relating to a hot-fix I did for adding `cadenceno` in __getitem__ of TargetPixelFile. The behavior works fine, but the test is wonky.

I ended up leaving in the possibility to toggle-off Hover.  I sort of like the option to disable the tool tips, which can themselves cause clutter at times.

**Comment by gully** [comment]
20180424_interact_cadences

**Comment by christinahedges** [comment]
If I change the quality bitmask to 'None', interact starts giving errors.

from lightkurve import KeplerTargetPixelFile
tpf = KeplerTargetPixelFile.from_archive(201572338, quality_bitmask=None)
tpf.interact()

**Comment by christinahedges** [comment]
You can add the human readable dates to the tool tips using:

tpf.timeobj.isot

Which returns an array of strings.

**Comment by christinahedges** [comment]
Short cadence files won't open.

**Comment by gully** [comment]
Working on improvement suggestions from 4/25/2018 from Geert and Christina:
- [ ] Anticipate possibility of `quality_flags=None` causing more cadences and NaN times
- [ ] Add human readable date in the hover-over tooltips using `tpf.timeobj.isot`
- [x] Turn off the `?` tooltip in the pixel data window
- [ ] Add a 💾 "Save" tooltip to the lightcurve window
- [ ] Improve short cadence performance
- [ ] Get interact to open from an IPython terminal shell (rather than just Jupyter)
- [ ] Add tests
- [ ] What happens to screen stretch if TPF is all zeros?
- [ ] Don't display anything in the TPF if there is no cadence available
- [ ] Try selection slider instead of intslider

**Comment by gully** [comment]
So getting interact to work for short cadence is very complicated.  A short cadence lightcurve can have 120,000 points, which is too large for Bokeh to handle without noticeable lag.  The widget slider motion grinds to a crawl in short cadence.  I ping-ed JJ Hermes to see if a short cadence redesign would be valuable for his science, and what design tradeoffs would be acceptable.  The easiest path for short cadence is likely a decimated, user-selected cadence range with a fixed number (~4000) of cadences.

**Comment by gully** [comment]
OK, I think this PR is ready to merge!!

**Comment by barentsen** [comment]
I have reviewed this PR and it looks ready to merge!  Please go ahead with a "squash-and-merge"!  The outstanding ideas can be migrated to a new issue or PR.  Thanks for making this happen @gully!

**Comment by barentsen** [comment]
:fireworks:

**Comment by gully** [comment]
Here's the gif animation for how it looks:
lightkurve_interact_8sec

**Pull request merged**